cli/command/volume/inspect_test.go

```diff
@@ -0,0 +1,150 @@
+package volume
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/cli/internal/test"
+	// Import builders to get the builder function as package function
+	. "github.com/docker/docker/cli/internal/test/builders"
+	"github.com/docker/docker/pkg/testutil/assert"
+	"github.com/docker/docker/pkg/testutil/golden"
+)
+
+func TestVolumeInspectErrors(t *testing.T) {
+	testCases := []struct {
+		args              []string
+		flags             map[string]string
+		volumeInspectFunc func(volumeID string) (types.Volume, error)
+		expectedError     string
+	}{
+		{
+			expectedError: "requires at least 1 argument",
+		},
+		{
+			args: []string{"foo"},
+			volumeInspectFunc: func(volumeID string) (types.Volume, error) {
+				return types.Volume{}, fmt.Errorf("error while inspecting the volume")
+			},
+			expectedError: "error while inspecting the volume",
+		},
+		{
+			args: []string{"foo"},
+			flags: map[string]string{
+				"format": "{{invalid format}}",
+			},
+			expectedError: "Template parsing error",
+		},
+		{
+			args: []string{"foo", "bar"},
+			volumeInspectFunc: func(volumeID string) (types.Volume, error) {
+				if volumeID == "foo" {
+					return types.Volume{
+						Name: "foo",
+					}, nil
+				}
+				return types.Volume{}, fmt.Errorf("error while inspecting the volume")
+			},
+			expectedError: "error while inspecting the volume",
+		},
+	}
+	for _, tc := range testCases {
+		buf := new(bytes.Buffer)
+		cmd := newInspectCommand(
+			test.NewFakeCli(&fakeClient{
+				volumeInspectFunc: tc.volumeInspectFunc,
+			}, buf),
+		)
+		cmd.SetArgs(tc.args)
+		for key, value := range tc.flags {
+			cmd.Flags().Set(key, value)
+		}
+		cmd.SetOutput(ioutil.Discard)
+		assert.Error(t, cmd.Execute(), tc.expectedError)
+	}
+}
+
+func TestVolumeInspectWithoutFormat(t *testing.T) {
+	testCases := []struct {
```

Comment:
Since the anonymous struct is re-used several times perhaps you could refactor it to its own named struct type  (unexported)